zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -219,44 +297,169 @@ where
             .map_err(Error::Builder)?;
     }
 
-    match to {
-        RecipientAddress::Shielded(to) => {
-            builder.add_sapling_output(ovk, to.clone(), value, memo.clone())
-        }
-
-        RecipientAddress::Transparent(to) => builder.add_transparent_output(&to, value),
+    for payment in request.payments() {
+        match &payment.recipient_address {
+            RecipientAddress::Shielded(to) => builder
+                .add_sapling_output(
+                    ovk,
+                    to.clone(),
+                    payment.amount,
+                    payment.memo.clone().unwrap_or_else(MemoBytes::empty),
```

Comment:
This uses an empty memo if none is provided; the `MemoRequired` error appears never to be used.